apps/taskman/api.py

```diff
@@ -0,0 +1,225 @@
+        """
+        unauthenticated health check API endpoint
+        """
+        logger.info("getting status")
+        status_code = 200 if self._is_healthy else 409
+        status_str = "ok" if self._is_healthy else "error"
+        return Response(data={"status": status_str}, status=status_code)
+
+
+@jira_token_description
+class task(GenericAPIView):
+    """
+    API endpoint for getting tasks by its key
+    """
+
+    @extend_schema(
+        responses=JiraIssueSerializer,
+    )
+    def get(self, request, task_key):
+        """Get a task from Jira given a task key"""
+        return JiraTaskmanQuerier(
+            token=request.headers.get("JiraAuthentication")
+        ).get_task(task_key)
+
+
+@jira_token_description
+class task_comment_new(GenericAPIView):
+    @extend_schema(
+        parameters=[
+            OpenApiParameter(name="content", required=True, type=str),
+        ],
+        description="Create a new comment in a task",
+        responses=JiraCommentSerializer,
+    )
+    def post(self, request, task_key):
+        serializer = TaskCommentSerializer(data=request.data)
+        serializer.is_valid(raise_exception=True)
+        return JiraTaskmanQuerier(
+            token=request.headers.get("JiraAuthentication")
+        ).create_comment(
+            issue_key=task_key,
+            body=request.data["content"],
+        )
+
+
+@jira_token_description
+class task_comment(GenericAPIView):
+    @extend_schema(
+        parameters=[
+            OpenApiParameter(name="content", required=True, type=str),
+        ],
+        description="Edit a comment in a task",
+        responses=JiraCommentSerializer,
+    )
+    def put(self, request, task_key, comment_id):
+        """Get a task from Jira given a task key"""
+        serializer = TaskCommentSerializer(data=request.data)
+        serializer.is_valid(raise_exception=True)
+        return JiraTaskmanQuerier(
+            token=request.headers.get("JiraAuthentication")
+        ).update_comment(
+            issue_key=task_key,
+            comment_id=comment_id,
+            body=serializer.validated_data["content"],
+        )
+
+
+@jira_token_description
+class task_group_new(GenericAPIView):
+    @extend_schema(
+        parameters=[
+            OpenApiParameter(name="name", required=True, type=str),
+            OpenApiParameter(name="description", type=str),
+        ],
+        description="Create a new group of tasks",
+        responses=JiraIssueSerializer,
+    )
+    def post(self, request):
+        serializer = TaskGroupSerializer(data=request.data)
+        serializer.is_valid(raise_exception=True)
+
+        return JiraTaskmanQuerier(
+            token=request.headers.get("JiraAuthentication")
+        ).create_group(
+            name=serializer.validated_data["name"],
+            description=serializer.validated_data["description"],
+        )
+
+
+@jira_token_description
+class task_group(GenericAPIView):
```

Comment:
Is there a reason why `task_group` and `task_group_new` need to be two separate classes? I am also not sure about the split of `task_comment` and `task_comment_new`.

Reply:
No problem then. I think that passing the key in the body would be worse. This is good :+1: I was just not clear on the split but when there is a reason then leave it as it is